vector.go

```diff
@@ -507,6 +518,35 @@ func (vec *vector) initArray(logicalType mapping.LogicalType, colIdx int) error
 	return nil
 }
 
+func (vec *vector) initUnion(logicalType mapping.LogicalType, colIdx int) error {
+	memberCount := int(mapping.UnionTypeMemberCount(logicalType))
+
+	if memberCount == 0 {
+		return addIndexToError(unsupportedTypeError("empty union"), colIdx)
+	}
+
+	vec.childVectors = make([]vector, memberCount)
+	for i := 0; i < memberCount; i++ {
+		memberType := mapping.UnionTypeMemberType(logicalType, mapping.IdxT(i))
+
+		err := vec.childVectors[i].init(memberType, colIdx)
+		mapping.DestroyLogicalType(&memberType)
+		if err != nil {
+			return err
+		}
+	}
+
+	vec.Type = TYPE_UNION
+	vec.getFn = func(vec *vector, rowIdx mapping.IdxT) any {
```

Comment:
Let's also add a "not implemented/ invalid code path" `panic` to `vec.setFn` (a dummy callback). That callback should be unreachable code. However, currently it is possible to trigger a panic/crash by trying to append to a `UNION` column (with the test below). Instead, we need to throw a `not implemented` there. Or, maybe, we also add support for `setFn` to this PR?

```Go
func TestAppenderUnion(t *testing.T) {
	c, db, conn, a := prepareAppender(t, `
	CREATE TABLE test (
		i UNION(num INTEGER, str VARCHAR)
	)`)
	defer cleanupAppender(t, c, db, conn, a)

	union := Union{
		Tag:   "num",
		Value: int32(42),
	}

	require.NoError(t, a.AppendRow(union))
	require.NoError(t, a.Flush())
}
```

Reply:
I agree, that it feels easier - or at least not much more work to just implement setting union types in the same PR. I gave it a try and implemented `setUnion`.